mava/types.py

```diff
@@ -94,18 +106,34 @@ class RNNEvalState(NamedTuple):
     key: chex.PRNGKey
     env_state: State
     timestep: TimeStep
+    step_count_: chex.Numeric
+    return_: chex.Numeric
     dones: chex.Array
-    hstate: chex.Array
-    step_count_: chex.Numeric = None
-    return_: chex.Numeric = None
+    hstate: HiddenState
+
+
+MavaState = TypeVar("MavaState", LearnerState, RNNLearnerState, EvalState, RNNEvalState)
 
 
-class ExperimentOutput(NamedTuple):
+class ExperimentOutput(NamedTuple, Generic[MavaState]):
```

Comment:
I think we need to double check this, I don't know if the error happen on my local machine only but this one give the following error {TypeError: Multiple inheritance with NamedTuple is not supported{

Reply:
Not sure but that seems to be a python 3.8 issue see [here](https://github.com/JustAnotherArchivist/snscrape/issues/111)

Reply:
They mentioned that they fixed it for python 3.9 however I am using 3.9 and still have it :(

Reply:
hmmmm I'm not sure then :thinking:

Reply:
I will try to look if there is another option to keep inheriting from both